server/resp/src/main/java/org/infinispan/server/resp/commands/string/MSETNX.java

```diff
@@ -39,12 +42,14 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
          return handler.myStage();
       }
       Log.SERVER.msetnxConsistencyMessage();
-      var entries = new HashMap<byte[], byte[]>();
+      // Using WBA so equals() ensures same key has only one entry with the last value in the map
+      var entriesWBA = new HashMap<WrappedByteArray, WrappedByteArray>();
       for (int i = 0; i < arguments.size(); i++) {
-         entries.put(arguments.get(i), arguments.get(++i));
+         entriesWBA.put(new WrappedByteArray(arguments.get(i)), new WrappedByteArray(arguments.get(++i)));
```

Comment:
Only the keys would need to be wrapped?

Reply:
oh yeah true, we can skip wrapping the value